tests/utils/tmpdir.c

```diff
@@ -0,0 +1,67 @@
+#define _XOPEN_SOURCE 500 /* nftw() is part of POSIX 500 spec*/
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <cmocka.h>
+// used to delete directory recursively
+#include <ftw.h>
+
+#include <stdlib.h>
+
+#include "utils/log.h"
+
+#include "tmpdir.h"
+
+// recursively delete folder using ftw
+static int rm_file(const char *pathname, const struct stat *sbuf, int type,
```

Comment:
Do you think it makes sense to add these nice files to `os.h`?

Reply:
Ehhhh, maybe. The problem is, if we add it to `os.h`, we'd also want to test every branch of it.

Basically, if it's not used in `src/`, we should probably keep the functions in `tests/`.